pkg/controller/datavolume-controller.go

```diff
@@ -2599,3 +2623,288 @@ func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
+	resourceRequirements, err := GetDefaultPodResourceRequirements(r.client)
+	if err != nil {
+		return nil, err
+	}
+
+	podName := sizeDetectionPodName(sourcePvc)
+	pod := &corev1.Pod{}
+	nn := types.NamespacedName{Namespace: sourcePvc.Namespace, Name: podName}
+	// Trying to get the pod if it already exists/create it if not
+	if err := r.client.Get(context.TODO(), nn, pod); err != nil {
+		if !k8serrors.IsNotFound(err) {
+			return nil, err
+		}
+		// Generate the pod spec
+		pod = makeSizeDetectionPodSpec(sourcePvc, r.client, r.pullPolicy)
+		if pod == nil {
+			return nil, errors.Errorf("Size detection pod could not be generated.")
+		}
+		if resourceRequirements != nil {
+			pod.Spec.Containers[0].Resources = *resourceRequirements
+		}
+		// Create the pod
+		if err := r.client.Create(context.TODO(), pod); err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				return nil, err
+			}
+		}
+
+		r.log.V(3).Info("Size-detection pod created\n", "pod.Name", pod.Name, "pod.Namespace", pod.Namespace)
+	}
+
+	return pod, nil
+}
+
+// makeSizeDetectionPodSpec creates and returns the full size-detection pod spec
+func makeSizeDetectionPodSpec(sourcePvc *corev1.PersistentVolumeClaim, client client.Client, pullPolicy string) *corev1.Pod {
+	image := os.Getenv("IMPORTER_IMAGE")
+	if image == "" {
+		return nil
+	}
+
+	workloadNodePlacement, err := GetWorkloadNodePlacement(client)
+	if err != nil {
+		return nil
+	}
+	// Generate individual specs
+	container := makeSizeDetectionContainerSpec(image, pullPolicy)
+	volume := makeSizeDetectionVolumeSpec(sourcePvc.Name)
+	objectMeta := makeSizeDetectionObjectMeta(sourcePvc, true, false)
+	// Assemble the pod
+	pod := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: *objectMeta,
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				*container,
+			},
+			Volumes: []corev1.Volume{
+				*volume,
+			},
+			RestartPolicy:     corev1.RestartPolicyOnFailure,
+			NodeSelector:      workloadNodePlacement.NodeSelector,
+			Tolerations:       workloadNodePlacement.Tolerations,
+			Affinity:          workloadNodePlacement.Affinity,
+			PriorityClassName: getPriorityClass(sourcePvc),
+		},
+	}
+
+	return pod
+}
+
+// makeSizeDetectionObjectMeta creates and returns the object metadata for the size-detection pod
+func makeSizeDetectionObjectMeta(sourcePvc *corev1.PersistentVolumeClaim, blockOwnerDeletion, isController bool) *metav1.ObjectMeta {
+	return &metav1.ObjectMeta{
+		Name:      sizeDetectionPodName(sourcePvc),
+		Namespace: sourcePvc.Namespace,
+		Annotations: map[string]string{
+			AnnCreatedBy: "yes",
+		},
+		Labels: map[string]string{
+			common.CDILabelKey:        common.CDILabelValue,
+			common.CDIComponentLabel:  common.ImporterPodName,
+			common.PrometheusLabelKey: common.PrometheusLabelValue,
+		},
+		OwnerReferences: []metav1.OwnerReference{
+			{
+				APIVersion:         "v1",
+				Kind:               "PersistentVolumeClaim",
```

Comment:
I would make the owner of this pod the data volume instead of the PVC. This way we can easily watch for pods that are owned by DVs and figure out which DV is associated with the pod. You can also an ownerRef easier like this. The datavolume controller has some examples.
```go
OwnerReferences = []metav1.OwnerReference{
  *metav1.NewControllerRef(dataVolume, schema.GroupVersionKind{
    Group:   cdiv1.SchemeGroupVersion.Group,
    Version: cdiv1.SchemeGroupVersion.Version,
    Kind:    "DataVolume",
  }),
}
```